### PR TITLE
feat: add a fuzzychooser to choose screen

### PR DESCRIPTION
**Pull request by Decodetalkers**
I want to add a fuzzy chooser to select screen

### CODE REVIEW

File: wayshot/src/wayshot.rs [hunk 1]
```diff
@@ -130,6 +147,26 @@ fn main() -> Result<(), Box<dyn Error>> {
         }
 
         capture_info.unwrap()
+    } else if args.get_flag("dialog") {
+        let outputs = output::get_all_outputs(&mut globals, &mut conn);
+        let outputnames: Vec<String> = outputs
```
Comment (Shinyzenith) [inline wayshot/src/wayshot.rs:152]: can we change this to `output_names`?

File: wayshot/src/wayshot.rs [hunk 2]
```diff
@@ -21,6 +21,9 @@ mod image_util;
 mod output;
 mod utils;
 
+use dialoguer::theme::ColorfulTheme;
+use dialoguer::FuzzySelect;
```
Comment (Shinyzenith) [inline wayshot/src/wayshot.rs:25]:
```suggestion
use dialoguer::{theme::ColorfulTheme, FuzzySelect};
```

File: wayshot/src/clap.rs [hunk 3]
```diff
@@ -57,4 +57,12 @@ pub fn set_flags() -> Command {
                 .conflicts_with("slurp")
                 .help("Choose a particular display to screenshot"),
         )
+        .arg(
+            arg!(--dialog)
+                .required(false)
+                .action(ArgAction::SetTrue)
+                .conflicts_with("slurp")
+                .conflicts_with("output")
+                .help("Choose a display with fuzzyselect dialog"),
```
Comment (Shinyzenith) [inline wayshot/src/clap.rs:66]:
```suggestion
                .help("Present a fuzzy selector for outputs"),
```

File: wayshot/src/clap.rs [hunk 4]
```diff
@@ -57,4 +57,12 @@ pub fn set_flags() -> Command {
                 .conflicts_with("slurp")
                 .help("Choose a particular display to screenshot"),
         )
+        .arg(
+            arg!(--dialog)
```
Comment (Shinyzenith) [inline wayshot/src/clap.rs:61]: would `--choose-output` be a better naming scheme?

File: wayshot/src/wayshot.rs [hunk 5]
```diff
@@ -130,6 +147,26 @@ fn main() -> Result<(), Box<dyn Error>> {
         }
 
         capture_info.unwrap()
+    } else if args.get_flag("dialog") {
```
Comment (Shinyzenith) [inline wayshot/src/wayshot.rs:150]: If we do change the flag name, then this line needs to be updated.

File: wayshot/src/clap.rs (hunk 4 above)
Comment (Decodetalkers) [reply wayshot/src/clap.rs]: but clap do not allow name have '-', emm

**Review by Shinyzenith — APPROVED** [review]

**Comment by Shinyzenith** [comment]
Thanks!

**Comment by Shinyzenith** [comment]
Merged as: https://github.com/waycrate/wayshot/commit/8e1633f6ef117ec9ab1c9d71c97c86bf3c6dfe2c

**Pull request merged**